data/dummy.js

```diff
@@ -0,0 +1,41 @@
+export default [
+  {
+    question: "What does HTML stand for?",
+    choices: [
+      "Hyper text markup language",
+      "Home tool markup language",
+      "Hyperlinks text markup language"
+    ],
+    correctAnswer: 1
+  },
```

Comment:
@AightA @kosenchiha Do you have have any thoughts on this structure for the questions and answers?

Reply:
I thought if we use data representing like below, it would be easier to iterate through it.
And I also suggest to save it in .json file, not .js
 
```
[
{
    "question": "What does HTML stand for?",
    "type": "radio",
    "answers": [
      {
        "answer": "Hyper text markup language",
        "flag": "true"
      },
      {
        "answer": "Home tool markup language",
        "flag": "false"
      },
      {
        "answer": "Hyperlinks text markup language",
        "flag": "false"
      },
      {
        "answer": "Hyperlinks text language",
        "flag": "false"
      }
    ]
  }
]
```

Reply:
Good suggestions! 

- `"type": "radio"` is a good idea since we will probably have multiple choice questions in which case we would have `"type": "checkbox"`
- Good to have a boolean associated with the answer to see whether the answer is correct. If we have `correctAnswer: 1` like before, then it's possible to add/remove/reorder the list of answers and forget to update `correctAnswer`. However, I would also suggest renaming the `flag` property to something more meaningful such as `isCorrect`
- Renaming to a `json` file is a good idea, since data we work with in the frontend is usually JSON which we normally get from a backend API
- @AightA also suggested on Slack `Do we need Question Id , and answers Id?`. This is a good suggestions as we will probably want to find specific questions and answers. We could do it with `questions.find(question => question.question === 'What does HTML stand for?')` but it's not ideal because if we change the question text then we need to change it here and also if the question is really long then it's not great to do a comparison with a really long string. Finding `id`s is better and also what is commonly done so let's add an `id` property.

Here's the suggested data structure. Can you update `data/dummy.js` with this structure @abdulkadiret ?
```
[
{
    "questionId": 1,
    "question": "What does HTML stand for?",
    "type": "radio",
    "answers": [
      {
        "answerId": 1,
        "answer": "Hyper text markup language",
        "isCorrect": "true"
      },
      {
        "answerId": 2,
        "answer": "Home tool markup language",
        "isCorrect": "false"
      },
      {
        "answerId": 3,
        "answer": "Hyperlinks text markup language",
        "isCorrect": "false"
      },
      {
        "answerId": 4,
        "answer": "Hyperlinks text language",
        "isCorrect": "false"
      }
    ]
  }
]
```

Reply:
Looking at this again, I think we should have this structure instead. Can you update data/dummy.js with this structure @abdulkadiret?

```
[
{
    "id": 1,
    "text": "What does HTML stand for?",
    "type": "radio",
    "answers": [
      {
        "id": 1,
        "text": "Hyper text markup language",
        "isCorrect": "true"
      },
      {
        "id": 2,
        "text": "Home tool markup language",
        "isCorrect": "false"
      },
      {
        "id": 3,
        "text": "Hyperlinks text markup language",
        "isCorrect": "false"
      },
      {
        "id": 4,
        "text": "Hyperlinks text language",
        "isCorrect": "false"
      }
    ]
  }
]
```

The reason is because if we have a variable like the following
```
const answer = {
  "answerId": 4,
  "answer": "Hyperlinks text language",
  "isCorrect": "false"
};
```
then when we access the variable's values we would write `answer.answerId` and `answer.answer` which is repetitive and not [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself). If we have the following

```
const answer = {
  "id": 4,
  "text": "Hyperlinks text language",
  "isCorrect": "false"
};
```
then we would write `answer.id`, `answer.text`, and `answer.isCorrect` which is much nicer to read.

cc @kosenchiha @AightA

Reply:
^^^Above you use "text" property name for two different properties, that is for the question and answer. Is that ok?

Reply:
@abdulkadiret Yes that's okay. The `text` property names are in different JS objects so that's fine.

If we have a questions variable, it would be something like this:
```javascript
const questions = [
  {
    "id": 1,
    "text": "What does HTML stand for?",
    "type": "radio",
    "answers": [
      {
        "id": 1,
        "text": "Hyper text markup language",
        "isCorrect": "true"
      },
    ],
  },
]
```
Accessing the question would be `questions[0].text` which reads nicely, and accessing an answer would be `questions[0].answers[0].text` which also reads nicely. Hope that answers your concerns.

If you think about how it would be stored in a relational database table, it might make more sense:
```
Question
-----------------
id      | integer
text    | varchar
type    | varchar
answers | integer
foreign key (answers) references answer(id)

Answer
--------------------
id         | integer
text       | varchar
is_correct | boolean
```